File: C/13._Roman_to_Integer.c
int romanToInt(char * s){
    int romanMap[26];
    romanMap['I'-'A'] = 1;
    romanMap['V'-'A'] = 5;
    romanMap['X'-'A'] = 10;
    romanMap['L'-'A'] = 50;
    romanMap['C'-'A'] = 100;
    romanMap['D'-'A'] = 500;
    romanMap['M'-'A'] = 1000;

    int ans = 0;
    int n = strlen(s);

    for(int i = 0 ; i < n ; i++){
        int value = romanMap[s[i]-'A'];
        if(i < n - 1 && value < romanMap[s[i+1]-'A']){
            ans -= value;
        }
        else{
            ans += value;
        }
    }
    return ans;
}

File: java/13._Roman_to_Integer.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int romanToInt(String s) {
        Map<Character,Integer> romanMap = new HashMap<>(){
            {
            put('I',1);
            put('V',5);
            put('X',10);
            put('L',50);
            put('C',100);
            put('D',500);
            put('M',1000);
            }
        };

        int res = 0;
        for(int i = 0 ; i < s.length() -1 ; i++){
            if (romanMap.get(s.charAt(i)) >= romanMap.get(s.charAt(i+1))){
                res += romanMap.get(s.charAt(i));
            }
            else{
                res -= romanMap.get(s.charAt(i));
            }
        }
        res += romanMap.get(s.charAt(s.length() -1));
        return res;
    }
}